### DISCUSSION TITLE
GOMP_parallel_start

### DISCUSSION THREAD

**Discussion by alexm98** [Homebrew/discussions#3668] [Writing Formulae/Casks] [answered]
Hello everyone, I am writing a formula for [GRASS GIS](https://grass.osgeo.org/).
In order to compile with OpenMP support, the software requires `GOMP_parallel_start` which is not found in `libomp`

A hack that I found to be able to do this is use includes from `libomp` but use GCC libs which include `GOMP_parallel_start`

```
"--with-openmp-includes=#{Formula["libomp"].opt_include}",
"--with-openmp-libs=#{Formula["gcc"].opt_lib}/gcc/current",
```

Needless to say, this is not a good idea. If I instead try to use the openmp-libs found in `libomp`:

```
"--with-openmp-libs=#{Formula["libomp"].opt_lib}",
```

I get the following message:

```
checking whether to use OpenMP... "yes"
checking for location of OpenMP includes... /opt/homebrew/opt/libomp/include
checking omp.h usability... yes
checking omp.h presence... yes
checking for omp.h... yes
checking for location of OpenMP library... /opt/homebrew/opt/libomp/lib
checking for GOMP_parallel_start... no
checking for GOMP_parallel_start in -lgomp... no
configure: error: *** Unable to locate OpenMP library.
```

What is a viable workaround for my formula that I could use? There does not seem to be a `libgomp` in homebrew-core.

**Comment by carlocab** [comment, marked as answer]
`libgomp` is provided by GCC, that's why this flag works:
```
"--with-openmp-libs=#{Formula["gcc"].opt_lib}/gcc/current"
```
You can use GCC's OpenMP implementation, but you should also use the matching headers that come with GCC.
